runtime/compiler/control/J9Options.cpp

```diff
@@ -1919,6 +2049,110 @@ J9::Options::fePreProcess(void * base)
+          (xxStartAsJITServerArgIndex > xxDisableStartAsJITServerArgIndex))
+         {
+         if (xxUseJITServerArgIndex > xxStartAsJITServerArgIndex) // client mode
+            {
+            compInfo->getPersistentInfo()->setRemoteCompilationMode(JITServer::CLIENT);
+
+            // parse -XX:+UseJITServer:server=<address/hostname>,port=<number> option if provided
+            char *xxUseJITServerOptionWithArgs = "-XX:+UseJITServer:"; // tail colon indicates server and port are provided
+            xxUseJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxUseJITServerOptionWithArgs, 0);
+            if (xxUseJITServerArgIndex >= 0)
+               {
+               // retrieve whole option part, i.e., server=<address/hostname>,port=<number>
+               char *options = NULL;
+               GET_OPTION_OPTION(xxUseJITServerArgIndex, ':', ':', &options);
+
+               char *scanLimit = options + strlen(options);
+               char delimiter = ',';
+               char *unRecognizedOptions = NULL;
+               while (options < scanLimit)
+                  {
+                  if (try_scan(&options, "server=")) // parse server=<address/hostname> option
+                     {
+                     char *address = scan_to_delim(PORTLIB, &options, delimiter);
+                     compInfo->getPersistentInfo()->setJITServerAddress(address);
+                     j9mem_free_memory(address);
+                     }
+                  else if (!JITServerParseOptionsCommon(&options, delimiter, compInfo, &unRecognizedOptions))
+                     break;
+                  }
+               }
+            }
+         else // server mode
+            {
+            compInfo->getPersistentInfo()->setRemoteCompilationMode(JITServer::SERVER);
+
+            // parse -XX:+StartAsJITServer:port=<number> option if provided
+            char *xxStartAsJITServerOptionWithArgs = "-XX:+StartAsJITServer:"; // tail colon indicates options are provided
+            xxStartAsJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxStartAsJITServerOptionWithArgs, 0);
+            if (xxStartAsJITServerArgIndex >= 0)
+               {
+               // retrieve whole option part, i.e., port=<number>
+               char *options = NULL;
+               GET_OPTION_OPTION(xxStartAsJITServerArgIndex, ':', ':', &options);
+
+               char *scanLimit = options + strlen(options);
+               char delimiter = ',';
+               char *unRecognizedOptions = NULL;
+               while (options < scanLimit && JITServerParseOptionsCommon(&options, delimiter, compInfo, &unRecognizedOptions));
+               }
+            }
+         }
+      if (compInfo->getPersistentInfo()->getRemoteCompilationMode() != JITServer::NONE)
+         {
+         // generate a random identifier for this JITServer instance.
+         // TODO: prevent collisions with some kind of atomic registration algo!
```

Comment:
Should the condition check here 
```
      if (compInfo->getPersistentInfo()->getRemoteCompilationMode() != JITServer::NONE)
```
be replaced with 
```
      if (compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT)
```
Otherwise `_clientUID` is set for `JITServer::SERVER` mode as well.

Reply:
Yes, we don't use an identifier at the server.